db/music.dbml

```diff
@@ -0,0 +1,299 @@
+Table submission {
+  submission_id uuid [pk, default: `gen_random_uuid()`]
+  participant_id int [not null, ref: > participant.participant_id]
+  bracket_id int [not null, ref: > bracket.bracket_id]
+  song_id int [not null, ref: > song.song_id]
+  order int [not null]
+  date_submitted timestamp [not null, default:`now()`]
+  active bool_int [not null]
+  is_dupe bool_int [not null]
+  title text [not null]
+  version text [null]
+  artist text [null]
+  collaborators text [null]
+  composer text [null]
+  link text [not null]
+  genre text [null]
+  emoji text [null]
+  comments text [null]
+
+  indexes {
+    (participant_id, bracket_id, song_id) [unique]
+  }
+}
+
+Table submission_history {
+  submission_history_id uuid [pk, default: `gen_random_uuid()`]
+  date_update timestamp [not null]
+  note_update text [not null, Note: 'dupe ineligible withdrawn "play-in loser"' ]
+  submission_id int [not null, ref: > submission.submission_id]
+  participant_id int [not null, ref: > participant.participant_id]
+  bracket_id int [not null, ref: > bracket.bracket_id]
+  song_id int [not null, ref: > song.song_id]
+  order int [not null]
+  date_submitted timestamp [not null, default:`now()`]
+  active bool_int [not null]
+  is_dupe bool_int [not null]
+  title text [not null]
+  version text [null]
+  artist text [null]
+  collaborators text [null]
+  composer text [null]
+  link text [not null]
+  genre text [null]
+  emoji text [null]
+  comments text [null]
+
+  indexes {
+    submission_id
+    participant_id
+    bracket_id
+    song_id
+  }
+
+  Note: 'composite unique index on submission_id, date_function (https://stackoverflow.com/a/26304364/2506158)'
+}
+
+Table link_relation {
+  link_realtion_id uuid [pk, default: `gen_random_uuid()`]
+  relation_id int [not null]
+  link_id int [not null, ref: > link.link_id]
+  type_link text [not null, Note: 'submission song seed' ]
+
+  indexes {
+    relation_id
+  }
+}
+
+Table link {
+  link_id uuid [pk, default: `gen_random_uuid()`]
+  url text [not null]
+  region text[] [null, Note: 'US Canada Japan Australia Europe Asia "North America" "South America" Africa Worldwide' ]
+}
+
+Table song {
+  song_id uuid [pk, default: `gen_random_uuid()`]
+  title text [not null]
+  title_english text [null]
+  version text [null]
+  artist text [not null, Note: 'CALCULATED']
+  collaborators text [null, Note: 'CALCULATED']
+  composer text [null]
+  year_min int [null]
+  year_max int [null]
+
+  indexes {
+    (title, version, artist, collaborators) [unique]
+  }
+}
+
+Table song_relation {
+  song_relation_id int [pk,increment]
+  song_from_id int [not null]
+  song_to_id int [not null]
+  relation_type text [not null, Note: 'cover live' ]
+}
+
+Table song_artist {
+  song_artist_id uuid [pk, default: `gen_random_uuid()`]
+  song_id int [not null, ref: > song.song_id]
+  artist_id int [not null, ref: > artist.artist_id]
+  credit_type text [not null, Note: 'primary collaborator original remixer composer' ]
```

Comment:
What is `original` supposed to represent here? Is that like from a cover? Or would that be more "This was a Talking Heads song and now David Byrne did a solo version of it, so David Byrne is the `primary` and Talking Heads is the `original`"?

Reply:
I think the idea was sort of a substitute for parent/child. Like if it's a cover, you can record the original artist that way. Not sure if it's really useful or not.